snap/snapcraft.yaml

```diff
@@ -14,6 +14,12 @@ layout:
   /usr/share/locale:
     bind: $SNAP/usr/share/locale
 
+plugs:
+  read-snap-data:
+    interface: system-files
```

Comment:
We could use the desktop-launch interface to get access to the icon.  Even though we aren't currently using that interface to launch snaps, it does provides access to the desktop files and icons.

Reply:
I'll test that.

Reply:
Mmm... it's not enough... the icon paths that I receive are inside the snaps, while "desktop-launch" only gives access to /var/lib/snapd/desktop/icons/...

Reply:
The desktop-launch interface does include "/snap/*/*/** r," and I think it's more compelling to get permission to plug desktop-launch than system-files.

https://github.com/snapcore/snapd/blob/master/interfaces/builtin/desktop_launch.go#L46

Reply:
Odd... it works, but I think that the last time that I tested it, it didn't... Maybe I forgot to connect the plug... Anyway, change done.